src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java

```diff
@@ -105,4 +115,36 @@
     }
     return mappingProfileDao.delete(mappingProfileId, tenantId);
   }
+
+  @Override
+  public Future<Void> validate(MappingProfile mappingProfile, OkapiConnectionParams params) {
+    if (CollectionUtils.isNotEmpty(mappingProfile.getTransformations())) {
+       return transformationFieldsService.getTransformationFields(params)
+        .compose(transformationFieldCollection -> {
+          List<TransformationField> transformationFields = transformationFieldCollection.getTransformationFields();
+          for (Transformations transformation : mappingProfile.getTransformations()) {
+            String fieldId = transformation.getFieldId();
+            if (StringUtils.isBlank(fieldId)) {
+              throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, "Field id is missing for mapping profile transformation");
```

Comment:
can we append transformation.fieldId to the message

Reply:
In this case it's blank,  so we cannot